File: src/api/ical.ts
import type { SessionHandle } from "~/models";

export const timetableICalURL = (session: SessionHandle, iCalToken: string, fileName = "timetable"): string => {
  const version = session.instance.version.join(".");
  return `${session.information.url}/ical/${fileName}.ics?icalsecurise=${iCalToken}&version=${version}&param=266f3d32`;
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/helpers/ical.kt
package ink.literate.pawnote.api.helpers

import ink.literate.pawnote.models.SessionHandle

fun timetableICalURL(
    session: SessionHandle,
    iCalToken: String,
    fileName: String = "timetable"
): String {
  val version = session.instance.version.joinToString(".")
  return "${session.information.url}/ical/${fileName}.ics?icalsecurise=${iCalToken}&version=${version}&param=266f3d32"
}
